File: aidlServer/src/main/java/com/example/aidl/MainActivity.java
package com.example.aidl;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
//adb pull data/anr/traces.txt ./mytraces.txt

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}

File: kotlin_mvvm_demo/src/main/java/com/example/kotlin_mvvm_demo/view/MainActivity.kt
package com.example.kotlin_mvvm_demo.view

import android.databinding.DataBindingUtil
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.example.kotlin_mvvm_demo.R
import com.example.kotlin_mvvm_demo.databinding.ActivityMainBinding
import com.example.kotlin_mvvm_demo.model.Animal
import com.example.kotlin_mvvm_demo.viewmodel.AnimalViewModel

class MainActivity : AppCompatActivity() {

    var binding:ActivityMainBinding ?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        var animalViewModel = AnimalViewModel(Animal("dog", 0))
        binding!!.animalVm = animalViewModel

    }
}
